packages/astro/src/cli/index.ts

```diff
@@ -95,24 +48,35 @@ async function printVersion() {
 	console.log(pkgVersion);
 }
 
-/** Merge CLI flags & config options (CLI flags take priority) */
-function mergeCLIFlags(astroConfig: AstroConfig, flags: CLIState['options']) {
-	if (typeof flags.sitemap === 'boolean') astroConfig.buildOptions.sitemap = flags.sitemap;
-	if (typeof flags.site === 'string') astroConfig.buildOptions.site = flags.site;
-	if (typeof flags.port === 'number') astroConfig.devOptions.port = flags.port;
-	if (typeof flags.hostname === 'string') astroConfig.devOptions.hostname = flags.hostname;
-	if (typeof flags.experimentalStaticBuild === 'boolean') astroConfig.buildOptions.experimentalStaticBuild = flags.experimentalStaticBuild;
-	if (typeof flags.drafts === 'boolean') astroConfig.buildOptions.drafts = flags.drafts;
+/** Determine which command the user requested */
+function resolveCommand(flags: Arguments): CLICommand {
+	if (flags.version) {
+		return 'version';
+	} else if (flags.help) {
+		return 'help';
+	}
+	const cmd = flags._[2];
+	switch (cmd) {
+		case 'dev':
+			return 'dev';
+		case 'build':
+			return 'build';
+		case 'preview':
+			return 'preview';
+		case 'check':
+			return 'check';
```

Comment:
Is `switch` necessary here? Why not something like...

```js
const validCommands = new Set(['dev', 'build', 'preview', 'check']);
if (validCommands.has(cmd)) return cmd;
return 'help'
```

Reply:
done!